apisix/admin/plugins.lua

```diff
@@ -85,16 +102,37 @@ function _M.get(name)
 end
 
 
-function _M.get_plugins_list()
-    local plugins = core.config.local_conf().plugins
+function _M.get_plugins_list(subsystem)
+    local http_plugins
+    local stream_plugins
+    if subsystem == "stream" then
+        stream_plugins = core.config.local_conf().stream_plugins
+    end
+    if subsystem == "http" then
+        http_plugins = core.config.local_conf().plugins
+    end
+
     local priorities = {}
     local success = {}
-    for i, name in ipairs(plugins) do
-        local plugin_name = "apisix.plugins." .. name
-        local ok, plugin = pcall(require, plugin_name)
-        if ok and plugin.priority then
-            priorities[name] = plugin.priority
-            table_insert(success, name)
+    if http_plugins then
+        for i, name in ipairs(http_plugins) do
+            local plugin_name = "apisix.plugins." .. name
+            local ok, plugin = pcall(require, plugin_name)
+            if ok and plugin.priority then
+                priorities[name] = plugin.priority
+                table_insert(success, name)
+            end
+        end
+    end
+
+    if stream_plugins then
+        for i, name in ipairs(stream_plugins) do
+            local plugin_name = "apisix.stream.plugins." .. name
+            local ok, plugin = pcall(require, plugin_name)
+            if ok and plugin.priority then
+                priorities[name] = plugin.priority
+                table_insert(success, name)
```

Comment:
ditto

Reply:
Those functions are to get the entire plugin loaded with plugin.load(). Here we just need the names.

Reply:
done